### ISSUE TITLE
no-redeclare false positives in class methods

### ISSUE THREAD

**Issue by lucacasonato** [denoland/deno_lint#856]
### Lint Name

`no-redeclare`

### Code Snippet

```ts
export class X {
  y() {
    const _res = "foo";
  }

  z() {
    const _res = "foo";
  }
}
```

### Expected Result

No `no-redeclare` hit.

### Actual Result

`no-redeclare` triggers on `const _res` in the `z()` method.

### Version

deno 1.13.2+b31dad8

**Comment by magurotuna** [comment]
@lucacasonato thanks for reporting. We've got the same kind of report in Discord as well, but I'm not able to reproduce it by running `deno lint` command in the specified canary build.
The error you encountered only appears in VSCode, right?

**Comment by magurotuna** [comment]
From a quick bisecting I found this issue began since https://github.com/denoland/deno/pull/11780
@dsherret do you have any idea?

**Comment by magurotuna** [comment]
@lucacasonato 

> Should I move the issue to denoland/deno repo?

Yup, I think that would be good. It's most likely due to the caching logic, not a linter issue.
